ratelimit/redis.go

```diff
@@ -77,31 +77,27 @@ func (c *clusterLimitRedis) measureQuery(format, groupFormat string, fail *bool,
 	c.metrics.MeasureSince(key, start)
 }
 
-func (c *clusterLimitRedis) startSpan(ctx context.Context, spanName string) func(bool) {
-	nop := func(bool) {}
+func parentSpan(ctx context.Context) (opentracing.Span, bool) {
 	if ctx == nil {
-		return nop
+		return nil, false
 	}
 
 	parentSpan := opentracing.SpanFromContext(ctx)
 	if parentSpan == nil {
-		return nop
+		return nil, false
 	}
 
-	span := c.ringClient.StartSpan(spanName, opentracing.ChildOf(parentSpan.Context()))
```

Comment:
I think we can keep this and maybe even common part as caller never uses parentSpan

Reply:
no, I don't want this. Who starts has to finish and this we can't do here anymore.

Reply:
Well, it is a member method of clusterLimitRedis named `startSpan` that starts it